### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by flassagne**
Handle Qml modules with qtcreator

### CODE REVIEW

File: install/cmut__install__install_qml.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+    install( DIRECTORY ${ARG__DIRECTORY}
+             DESTINATION ${ARG__DESTINATION}
+             COMPONENT ${ARG__COMPONENT}
+             FILES_MATCHING
+             PATTERN "*.qml"
+             PATTERN "qmldir" )
+
+     set_target_properties(${target} PROPERTIES CMUT__QML_DIRECTORIES ${ARG__DESTINATION})
```
Comment (ledocc) [inline install/cmut__install__install_qml.cmake]: CMUT__QML_DIRECTORIES, this is a list, so get previous list, add element, set list

File: install/cmut__install__install_qml.cmake [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    cmut__utils__parse_arguments(
+        cmut__install__install_qml
+        ARG_
+        ""
+        "DESTINATION;COMPONENT"
+        "DIRECTORY"
+        ${ARGN}
+    )
+
+    if(NOT ARG__DIRECTORY)
```
Comment (ledocc) [inline install/cmut__install__install_qml.cmake:15]: this one could not have default value

File: install/cmut__install__install_qml.cmake [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+             DESTINATION ${ARG__DESTINATION}
+             COMPONENT ${ARG__COMPONENT}
+             FILES_MATCHING
+             PATTERN "*.qml"
+             PATTERN "qmldir" )
+
+     set_target_properties(${target} PROPERTIES CMUT__QML_DIRECTORIES ${ARG__DESTINATION})
+
+     get_target_property(_export_properties ${target} EXPORT_PROPERTIES)
+     list(APPEND _export_properties CMUT__QML_DIRECTORIES)
```
Comment (ledocc) [inline install/cmut__install__install_qml.cmake]: test if CMUT__QML_DIRECTORIES is not already in EXPORT_PROPERTIES
```
if(NOT CMUT__QML_DIRECTORIES IN_LIST EXPORT_PROPERTIES)
...
endif()
```

File: install/cmut__install__install_qml.cmake (hunk 2 above)
Comment (flassagne) [reply install/cmut__install__install_qml.cmake:15]: Done

File: install/cmut__install__install_qml.cmake (hunk 1 above)
Comment (flassagne) [reply install/cmut__install__install_qml.cmake]: Done

File: install/cmut__install__install_qml.cmake (hunk 2 above)
Comment (flassagne) [reply install/cmut__install__install_qml.cmake:15]: Done

File: qtcreator/cmut__qtcreator__import_qml.cmake [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    set( QML_IMPORT_PATH "${_qml_directories}" CACHE STRING "Qml import path" FORCE)
+
+endfunction()
+
+
+function(__cmut__qtcreator__get_dependency_install_directory dependency_install_directory dependency)
+# PREDICATE Library directory is one delph children of install directory
+
+    if( ${CMAKE_BUILD_TYPE} STREQUAL "Release" )
+         get_target_property(_dependency_location ${dependency} IMPORTED_LOCATION_RELEASE)
```
Comment (ledocc) [inline qtcreator/cmut__qtcreator__import_qml.cmake]: try the current config, then try  the default config
```
string(TOUPPER ${CMAKE_BUILD_TYPE} BUILD_TYPE_UPPER)
get_target_property(_dependency_location ${dependency} IMPORTED_LOCATION_${CMAKE_BUILD_TYPE})
if(NOT _dependency_location)
    get_target_property(_dependency_location ${dependency} IMPORTED_LOCATION)
endif()
```

File: qtcreator/cmut__qtcreator__import_qml.cmake [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    set(_qml_directories)
+    foreach(_dependency IN LISTS target_dependencies)
+        get_target_property(_qml ${_dependency} CMUT__QML_DIRECTORIES)
+
+        if( NOT _qml)
+            continue()
+        endif()
+
+        __cmut__qtcreator__get_dependency_install_directory(_dependency_install_dir ${_dependency})
```
Comment (ledocc) [inline qtcreator/cmut__qtcreator__import_qml.cmake]: test if _dependency_install_dir exists
```
if(EXISTS /the/path/to/test)
    ...
endif()
```

File: qtcreator/cmut__qtcreator__import_qml.cmake [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+# PREDICATE Library directory is one delph children of install directory
+
+    if( ${CMAKE_BUILD_TYPE} STREQUAL "Release" )
+         get_target_property(_dependency_location ${dependency} IMPORTED_LOCATION_RELEASE)
+     else()
+         get_target_property(_dependency_location ${dependency} IMPORTED_LOCATION_DEBUG)
+    endif()
+
+    if( NOT _dependency_location )
+        cmut_error("We can't find your dependency location. Verify your build type (Release, Debug, Coverage)")
```
Comment (ledocc) [inline qtcreator/cmut__qtcreator__import_qml.cmake]: log like this:
```
cmut_error("[cmut][qtcreator][import_qml] - ...")
```

File: qtcreator/cmut__qtcreator__import_qml.cmake [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+# and propose code completion in multi libraries environment we have to export
+# QML files and set QML_IMPORT_PATH variable.
+
+function(cmut__qtcreator__import_qml target)
+
+    get_target_property(_target_dependencies ${target} LINK_LIBRARIES)
+
+    __cmut__qtcreator__get_qml_import_paths(_qml_directories "${_target_dependencies}")
+
+    set( QML_IMPORT_PATH "${_qml_directories}" CACHE STRING "Qml import path" FORCE)
```
Comment (ledocc) [inline qtcreator/cmut__qtcreator__import_qml.cmake]: set QML_IMPORT_PATH as INTERNAL data to not see it in cmake gui, thought?

File: install/cmut__install__install_qml.cmake (hunk 3 above)
Comment (flassagne) [reply install/cmut__install__install_qml.cmake]: Done

File: qtcreator/cmut__qtcreator__import_qml.cmake (hunk 4 above)
Comment (flassagne) [reply qtcreator/cmut__qtcreator__import_qml.cmake]: Done

File: qtcreator/cmut__qtcreator__import_qml.cmake (hunk 5 above)
Comment (flassagne) [reply qtcreator/cmut__qtcreator__import_qml.cmake]: done

File: qtcreator/cmut__qtcreator__import_qml.cmake (hunk 6 above)
Comment (flassagne) [reply qtcreator/cmut__qtcreator__import_qml.cmake]: Done

File: qtcreator/cmut__qtcreator__import_qml.cmake (hunk 7 above)
Comment (flassagne) [reply qtcreator/cmut__qtcreator__import_qml.cmake]: Done

File: install/cmut__install__install_qml.cmake [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    endif()
+
+    set_target_properties(${target} PROPERTIES CMUT__QML_DIRECTORIES "${_cmut__qml_directory}")
+
+endfunction()
+
+
+function(__cmut__export_qml_directories target)
+
+    if(NOT CMUT__QML_DIRECTORIES IN LIST EXPORT_PROPERTIES)
```
Comment (ledocc) [inline install/cmut__install__install_qml.cmake]: IN_LIST

Comment (flassagne) [reply install/cmut__install__install_qml.cmake]: Done

**Pull request merged**